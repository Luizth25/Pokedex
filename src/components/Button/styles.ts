import styled from "styled-components";

export const Button = styled.button`
  width: 140px;
  height: 40px;
  background-color: #ea5952;
  text-align: center;
  color: #000;
  font-size: 16px;
  border: 2px solid ${({ theme }) => theme.colors.borderColor};
  border-radius: 4px;
  cursor: pointer;
  padding: 10px;
`;
