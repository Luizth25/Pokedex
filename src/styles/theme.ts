declare module "styled-components" {
  export interface DefaultTheme {
    colors: typeof themeColors;
  }
}

const themeColors = {
  background: "#CC322A",
  textBackground: "#E6D596",
  titleColor: "#4C3434",
  infoColor: "#642C34",
  borderColor: "#A7B9D3",
};

const theme = {
  colors: themeColors,
};

export default theme;
