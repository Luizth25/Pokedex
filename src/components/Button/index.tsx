import { TButtonProps } from "./types";
import * as s from "./styles";

const Button = ({ label, onClick }: TButtonProps) => (
  <s.Button onClick={onClick}>{label}</s.Button>
);

export default Button;
