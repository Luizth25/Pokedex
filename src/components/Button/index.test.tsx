import { fireEvent, render, screen } from "@testing-library/react";
import { ThemeProvider } from "styled-components";

import Button from ".";

import theme from "styles/theme";

describe("The Button Component", () => {
  it("Should be defined", () => {
    expect.assertions(1);

    expect(Button).toBeDefined();
  });

  it("Should be Have a Label and clicked", () => {
    expect.assertions(2);

    const fn = jest.fn();

    render(
      <ThemeProvider theme={theme}>
        <Button label="test" onClick={fn} />
      </ThemeProvider>
    );

    const button = screen.getByRole("button", { name: "test" });
    fireEvent.click(button);

    expect(button).toBeInTheDocument();
    expect(fn).toBeCalledTimes(1);
  });
});
