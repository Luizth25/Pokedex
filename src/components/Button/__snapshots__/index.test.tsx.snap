// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Button Component Should match snapshot 1`] = `
<div>
  <button
    class="sc-bcXHqe hEKieS"
  >
    test
  </button>
</div>
`;
